api/handlers/service_plan.go

```diff
@@ -130,6 +132,22 @@ func (h *ServicePlan) updatePlanVisibility(r *http.Request) (*routing.Response,
 	return routing.NewResponse(http.StatusOK).WithBody(presenter.ForServicePlanVisibility(visibility, h.serverURL)), nil
 }
 
+func (h *ServicePlan) delete(r *http.Request) (*routing.Response, error) {
+	authInfo, _ := authorization.InfoFromContext(r.Context())
+	logger := logr.FromContextOrDiscard(r.Context()).WithName("handlers.service-plan.delete")
+
+	planGUID := routing.URLParam(r, "guid")
+	if _, err := h.servicePlanRepo.GetPlan(r.Context(), authInfo, planGUID); err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "failed to get plan: %s", planGUID)
+	}
+
+	if err := h.servicePlanRepo.DeletePlan(r.Context(), authInfo, planGUID); err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "failed to delete plan: %s", planGUID)
```

Comment:
We should use the `apierrors.ForbiddenAsNotFound` to prevent potential leaks of information for plan guids that the user is not authorized to see.

Reply:
Fair, let's leave it like that